### ISSUE TITLE
BagOfGold v4.5.4 - First time install error

### ISSUE THREAD

**Issue by C4BR3R4** [Rocologo/BagOfGold#142]
Errors in console to installing plugin:
`[BagOfGold] Enabling BagOfGold v4.5.3
[BagOfGold] Config.yml was backed up to plugins/BagOfGold/backup/config.yml.bak1
[BagOfGold][Debug] bStat is enabled
[BagOfGold][Debug] Initialize database
[BagOfGold] Database version 3 detected.
[BagOfGold]Copying players from BagOfGold til BagOfGoldCore database
Error occurred while enabling BagOfGold v4.5.3 (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "one.lindegaard.BagOfGold.storage.DataStoreManager.flush()" because the return value of "one.lindegaard.BagOfGold.BagOfGold.getDataStoreManager()" is null
        at one.lindegaard.BagOfGold.storage.SQLiteDataStore.migrateDatabaseLayoutFromV3ToV4(SQLiteDataStore.java:221) ~[BagOfGold-4.5.3.jar:?]
        at one.lindegaard.BagOfGold.storage.DatabaseDataStore.initialize(DatabaseDataStore.java:143) ~[BagOfGold-4.5.3.jar:?]
        at one.lindegaard.BagOfGold.BagOfGold.onEnable(BagOfGold.java:144) ~[BagOfGold-4.5.3.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:279) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:192) ~[paper-1.19.4.jar:git-Paper-510]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[paper-1.19.4.jar:git-Paper-510]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_19_R3.CraftServer.enablePlugin(CraftServer.java:555) ~[paper-1.19.4.jar:git-Paper-510]
        at org.bukkit.craftbukkit.v1_19_R3.CraftServer.enablePlugins(CraftServer.java:466) ~[paper-1.19.4.jar:git-Paper-510]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:638) ~[paper-1.19.4.jar:git-Paper-510]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:437) ~[paper-1.19.4.jar:git-Paper-510]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:308) ~[paper-1.19.4.jar:git-Paper-510]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1104) ~[paper-1.19.4.jar:git-Paper-510]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.19.4.jar:git-Paper-510]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
[BagOfGold] Disabling BagOfGold v4.5.3`

**Comment by Rocologo** [comment]
First time? But it looks like it's trying to upgrade an old database? 

You can stop your server and then delete bagofgold.db and set database_version in config.yml to 0 and then start the server again

**Comment by C4BR3R4** [comment]
Testing...

**Comment by C4BR3R4** [comment]
> First time? But it looks like it's trying to upgrade an old database?
> 
> You can stop your server and then delete bagofgold.db and set database_version in config.yml to 0 and then start the server again

Is first time. 

I have deleted the bagofgold.db file and the customitemslib.db file and set database_version in config.yml to 0 in both plugins and the problem continues.

**Comment by Rocologo** [comment]
Did you stop the server before changing the value to 0?

**Comment by C4BR3R4** [comment]
Log: https://pastebin.com/ZUmEWsXz

**Comment by Rocologo** [comment]
This one 

one.lindegaard.BagOfGold.storage.SQLiteDataStore.migrateDatabaseLayoutFromV3ToV4(SQLiteDataStore.java:221)

Means that the plugin try to update the database, which should not happen. If you have deleted the DB file you can also try to set version to 4. (Newest)

**Comment by C4BR3R4** [comment]
> 

Changed to 4 deleting the database and the problem continues.

**Comment by Rocologo** [comment]
I have created a copy of my test server and deleted the db files from both CustomItemsLib, BagOfGold and Mobhunting and started my test server. No errors???

Then I checked your log file and the database_version is 3 not 4

[18:10:05] [Server thread/INFO]: [BagOfGold] Database version 3 detected.


1) Stop the server
2) Change database_version to 4
3) Start the server

**Comment by Rocologo** [comment]
Or try snapshot 131 or newer

https://jenkins.lindegaard.one/job/BagOfGold/

**Comment by C4BR3R4** [comment]
I have deleted the entire folders. Installed BagOfGold 4.5.5 #132 and Mobhunting.
I keep getting errors now about BagOfGold and also Mobhunting.

Log server: https://pastebin.com/NtAJ5NTy

**Comment by Rocologo** [comment]
I so strange why you are having so many problems. Are you running your server on Linux or Windows ? 

This error

[13:47:03] [Server thread/ERROR]: Error occurred while enabling MobHunting v8.5.3 (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "one.lindegaard.MobHunting.storage.DataStoreManager.flush()" because the return value of "one.lindegaard.MobHunting.MobHunting.getDataStoreManager()" is null

Should be fixed in newest snapshot from yesterday: https://jenkins.lindegaard.one/job/MobHunting/

Yesterday I only tested what happenden when I deleted the db file. I will try to delete the folders completely later today and see  if I can get this error too:

[13:47:03] [Server thread/INFO]: [MobHunting] Language file does not exist. Creating a new file based on en_US. You need to translate the file yourself.
[13:47:03] [Server thread/WARN]: java.io.IOException: No such file or directory
[13:47:03] [Server thread/WARN]:    at java.base/java.io.UnixFileSystem.createFileExclusively(Native Method)

**Comment by C4BR3R4** [comment]
Server is running "Linux 5.4.0-90-generic (amd64)" under Java JRE 18. Hosting by "Sparkedhost.com" and "Pterodactyl Panel".

Until before the last test I did not have MobHunting installed, I understand that it is not a dependency?
But I have only been able to install BagOfGold correctly after installing MobHunting and the last mentioned error.

Then I have restarted the server again without MobHunting and the error did not appear again.
